.github/workflows/tests.yml

```diff
@@ -0,0 +1,50 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+      - 'stable/**'
+  pull_request:
+    branches:
+      - master
+      - 'stable/**'
+
+jobs:
+  tests:
+    name: ${{matrix.extra_name}}${{matrix.sim}} (${{matrix.sim-version}}) | ${{matrix.os}} | Python ${{matrix.python-version}} ${{matrix.may_fail && '| May Fail' || ''}}
+    runs-on: ${{matrix.os}}
+    env:
+      SIM: ${{matrix.sim}}
+      TOPLEVEL_LANG: ${{matrix.lang}}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - sim: icarus
+          sim-version: apt
+          lang: verilog
+          python-version: 3.8
```

Comment:
Maybe we want to go with 3.6 here to make sure no newer features creep in accidentally?

Reply:
```suggestion
          python-version: 3.6
```